### PR TITLE
Stops explosion distant sounds from being global

### LINKED ISSUE #4578: Explosion Sounds are Global
## Description
When some miner dingus crashes their ship into an asteroid, everyone still on the station hears it.

### Steps to Reproduce

1. ask someone to take a shuttle and crash it into a faraway asteroid  
2. wait for it to happen
3. hear the explosion sound, even though you're nowhere near it.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
fixes #4578
CL: [Fix] fixed explosion distant sound being global

### CODE REVIEW

**Comment by Bod9001** [comment]
surely you would need to increase the range?

**Comment by MaxIsJoe** [comment]
> surely you would need to increase the range?

I honestly have no idea what's the "good" range but from my testing on the teleporter maze station it seemed ok-ish

**Comment by corp-0** [comment]
we can tweak numbers later, I'm more interested in knowing if it works as expected

**Comment by MaxIsJoe** [comment]
Tested, station seems to be silent when the shuttle hits.

**Comment by MaxIsJoe** [comment]
https://youtu.be/D61y5l51fn8

**Review by corp-0 — DISMISSED** [review]

**Comment by Bod9001** [comment]
it's easy to revert if causes issues

**Comment by ThatDan123** [comment]
Making this not global will mean that the sound is only heard by players within 15 tiles, see ServerMessage SendToNearbyPlayers(). 

The global bool also affects whether the sound is muffled.

The issue is more likely the sound prefab in the addressables project distance being too big, or having the wrong audio type (2d vs 3d)

**Comment by corp-0** [comment]
> Making this not global will mean that the sound is only heard by players within 15 tiles, see ServerMessage SendToNearbyPlayers().
> 
> The global bool also affects whether the sound is muffled.
> 
> The issue is more likely the sound prefab in the addressables project distance being too big, or having the wrong audio type (2d vs 3d)

Explosion code has another distance check, not related to sound distance check from the sound prefab, to handle explosion Vs station groan sound. The issue is it was always playing the groan. It was also not spacial and you couldn't determine the position of the explosion by its sound which is bad because it could have happened in lavalan and you would hear it in the station

**Comment by Bod9001** [comment]
where is that code that checks if it's explosion?

**Comment by corp-0** [comment]
It is the only method this class has

**Comment by Bod9001** [comment]
Wouldn't there be a check on SoundManager.PlayNetworkedAtPosAsync?

**Comment by corp-0** [comment]
Why there? This is only relevant for explosions

**Comment by Bod9001** [comment]
PlayNetworkedAtPosAsync -> global  = False -> PlaySoundMessage.SendToNearbyPlayers -> SendToNearbyPlayers ->  if (Vector2.Distance(worldPosition, players[i].GameObject.transform.position) > 15f)

**Comment by corp-0** [comment]
adding "on hold" label until we find a good solution for the hard coded 15 tiles limit

**Comment by MaxIsJoe** [comment]
This should be fixed now

**Review by ThatDan123 — APPROVED** [review]

File: UnityProject/Assets/Scripts/Items/Weapons/ExplosionUtils.cs [hunk 1]
```diff
@@ -26,20 +23,22 @@ public static class ExplosionUtils
...
+			if (shakeDistance > distantSound.AudioSource.maxDistance)
 			{
-				AddressableAudioSource distantName = DISTANT_EXPLOSION_SOUNDS[Random.Range(0, DISTANT_EXPLOSION_SOUNDS.Length)];
-				_ = SoundManager.PlayNetworkedAtPosAsync(distantName, worldPosition, global: true);
+				_ = SoundManager.PlayNetworkedAtPosAsync(distantSound, worldPosition, global: true);
 			}
-			if (shakeIntensity > GROAN_THRESHOLD)
+			if (shakeIntensity > groanSound.AudioSource.maxDistance)
 			{
-				AddressableAudioSource groanName = STATION_GROAN_SOUNDS[Random.Range(0, STATION_GROAN_SOUNDS.Length)];
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Items/Weapons/ExplosionUtils.cs:41]: wouldn't this still play across matrices?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Weapons/ExplosionUtils.cs:41]: hm? wdym?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Weapons/ExplosionUtils.cs:41]: if you're on an asteroid You will still hear this because it's global

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Weapons/ExplosionUtils.cs:41]: Daniel explained this ``global`` variable is just misnamed. ``global`` in this context means "everyone spawns the prefab" but the falloff distance set in the prefab still applies.

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Weapons/ExplosionUtils.cs:41]: ok cool

**Comment by Bod9001** [comment]
just noting it bypasses the range check

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**